geomesa-hbase/geomesa-hbase-spark/src/main/scala/org/locationtech/geomesa/spark/hbase/GeoMesaHBaseInputFormat.scala

```diff
@@ -0,0 +1,157 @@
+import org.apache.hadoop.io.Text
+import org.apache.hadoop.mapreduce._
+import org.geotools.data.DataStoreFinder
+import org.geotools.filter.identity.FeatureIdImpl
+import org.geotools.filter.text.ecql.ECQL
+import org.geotools.process.vector.TransformProcess
+import org.locationtech.geomesa.hbase.data.{HBaseDataStore, HBaseFeature}
+import org.locationtech.geomesa.hbase.index.HBaseFeatureIndex
+import org.locationtech.geomesa.index.api.GeoMesaFeatureIndex
+import org.locationtech.geomesa.jobs.GeoMesaConfigurator
+import org.locationtech.geomesa.utils.index.IndexMode
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+import org.opengis.filter.Filter
+
+import scala.collection.JavaConversions._
+
+/**
+  * Input format that allows processing of simple features from GeoMesa based on a CQL query
+  */
+class GeoMesaHBaseInputFormat extends InputFormat[Text, SimpleFeature] with LazyLogging {
+
+  val delegate = new MultiTableInputFormat
+
+  var sft: SimpleFeatureType = _
+  var table: GeoMesaFeatureIndex[HBaseDataStore, HBaseFeature, Mutation] = _
+
+  class HBaseGeoMesaRecordReader(sft: SimpleFeatureType,
+                                 table: HBaseFeatureIndex,
+                                 reader: RecordReader[ImmutableBytesWritable, Result],
+                                 hasId: Boolean,
+                                 filterOpt: Option[Filter],
+                                 transformSchema: Option[SimpleFeatureType]
+                                ) extends RecordReader[Text, SimpleFeature] {
+
+
+    private def nextFeatureFromOptional(fn: Result => Option[SimpleFeature]) = () => {
+      staged = null
+      while(reader.nextKeyValue() && staged == null) {
+        fn(reader.getCurrentValue) match {
+          case Some(feature) =>
+            staged = feature
+
+          case None =>
+            staged = null
+        }
+      }
+    }
+
+    private def nextFeatureFromDirect(fn: Result => SimpleFeature) = () => {
+      staged = null
+      while(reader.nextKeyValue() && staged == null) {
+        staged = fn(reader.getCurrentValue)
+      }
+    }
+
+
+    var staged: SimpleFeature = _
+    private val nextFeature =
+      (filterOpt, transformSchema) match {
+        case (Some(filter), Some(ts)) =>
+          val indices = ts.getAttributeDescriptors.map { ad => sft.indexOf(ad.getLocalName) }
+          val fn = table.toFeaturesWithFilterTransform(sft, filter, Array.empty[TransformProcess.Definition], indices.toArray, ts)
+          nextFeatureFromOptional(fn)
+
+        case (Some(filter), None) =>
+          val fn = table.toFeaturesWithFilter(sft, filter)
+          nextFeatureFromOptional(fn)
+
+        case (None, Some(ts))         =>
+          val indices = ts.getAttributeDescriptors.map { ad => sft.indexOf(ad.getLocalName) }
+          val fn = table.toFeaturesWithTransform(sft, Array.empty[TransformProcess.Definition], indices.toArray, ts)
```

Comment:
I think I want to punt on that till a later ticket.  Since we don't push down the transformations in HBase or Bigtable, they can be handled effectively by Spark.  However, when we start leveraging filters and coprocessors, I think we should revisit this.